src/pipx/commands/common.py

```diff
@@ -172,3 +178,104 @@ def _get_list_output(
             f"    - {red(name)} (symlink missing or pointing to unexpected location)"
         )
     return "\n".join(output)
+
+
+def _package_name_from_spec(
+    package_spec: str, python: str, *, pip_args: List[str], verbose: bool
+) -> str:
+    start_time = time.time()
+
+    # shortcut if valid PyPI name and not a local path
+    if valid_pypi_name(package_spec) and not Path(package_spec).exists():
+        package_name = package_spec
+        logging.info(f"Determined package name: {package_name}")
+        logging.info(f"Package name determined in {time.time()-start_time:.1f}s")
+        return package_name
+
+    with tempfile.TemporaryDirectory() as temp_venv_dir:
+        venv = Venv(Path(temp_venv_dir), python=python, verbose=verbose)
+        venv.create_venv(venv_args=[], pip_args=[])
+        package_name = venv.install_package_no_deps(
+            package_or_url=package_spec, pip_args=pip_args
+        )
+
+    logging.info(f"Package name determined in {time.time()-start_time:.1f}s")
+    return package_name
+
+
+def _run_post_install_actions(
```

Comment:
This should probably no longer begin with an underscore since we are referencing it from outside of this module.

Reply:
6cda48b